create.go

```diff
@@ -190,6 +221,22 @@ func (r *Resource) computeCreateEventPatches(ctx context.Context, accessor metav
 		}
 	}
 
+	// Check all node versions held by the cluster status and add the version the
+	// guest cluster successfully migrated to, to the historical list of versions.
+	{
+		isUpdated := clusterStatus.HasUpdatedCondition()
+		sameCount := currentNodeCount != 0 && currentNodeCount == desiredNodeCount
+		sameVersion := allNodesHaveVersion(clusterStatus.Nodes, desiredVersion)
+
+		if isUpdated && sameCount && sameVersion {
+			patches = append(patches, Patch{
+				Op:    "replace",
+				Path:  "/status/cluster/versions",
+				Value: clusterStatus.WithNewVersion(desiredVersion),
+			})
+		}
+	}
```

Comment:
This got just moved more to the bottom so the condition management above is a bit more telling about the natural status transitions.